### PR TITLE
update HLT `addOnTests` for Run-3 Data with 2021 pilot-beam data

### PR DESCRIPTION
**Pull request by missirol**
#### PR description:

This PR updates the configuration of the HLT `addOnTests` for (Run-3) collisions Data.

Up to now, these tests still had to run on Run-2 Data, using ad-hoc customisations needed to cope with the `Run3` era modifier.

This PR changes the input EDM file to one from the 2021 pilot-beam test, updates the dictionary of HLT GTs in `autoCondHLT` to use the Run-3 GTs (instead of the Run-2 ones), and removes the aforementioned ad-hoc customisations where needed. 

*Caveat* : the test for the `HIon` menu is also made to run on pilot-beam (pp) data, given the lack of HI collisions data post-Run2. The standard customisation for HI to change the name of the RAW-data input collection thus has to be disabled (it is commented out, not removed). Whenever new HI collisions data becomes available, the inputs for the relevant `HIon` tests will have to be updated accordingly, as well as the value of `run3_data_HIon` in `autoCondHLT`.

#### PR validation:

TSG tests (including `runAll.csh dev` and `addOnTests.py`).

#### If this PR is a backport, please specify the original PR and why you need to backport that PR:

N/A

### CODE REVIEW

**Comment by missirol** [comment]
please test

File: Configuration/HLT/python/autoCondHLT.py [hunk 1]
```diff
@@ -43,21 +43,21 @@
...
-    'run3_data_FULL'         : ('run2_data_relval'        ,l1Menus['FULL']),
-    'run3_data_GRun'         : ('run2_data_relval'        ,l1Menus['GRun']),
-    'run3_data_HIon'         : ('run2_data_promptlike_hi' ,l1Menus['HIon']),
-    'run3_data_PIon'         : ('run2_data_relval'        ,l1Menus['PIon']),
-    'run3_data_PRef'         : ('run2_data_relval'        ,l1Menus['PRef']),
+    'run3_data_FULL'         : ('run3_data'               ,l1Menus['FULL']),
+    'run3_data_GRun'         : ('run3_data'               ,l1Menus['GRun']),
+    'run3_data_HIon'         : ('run3_data'               ,l1Menus['HIon']),
+    'run3_data_PIon'         : ('run3_data'               ,l1Menus['PIon']),
+    'run3_data_PRef'         : ('run3_data'               ,l1Menus['PRef']),
```
Comment (Martin-Grunewald) [inline Configuration/HLT/python/autoCondHLT.py:60]: What is the reason to use run3_data instead of run3_data_prompt?

Comment (missirol) [reply Configuration/HLT/python/autoCondHLT.py:60]: No reason.. was an oversight on my part, sorry. Will change to `run3_data_prompt`.

**Comment by missirol** [comment]
please test

(force-pushed to minimise mistakes in the history)

**Comment by missirol** [comment]
Just noticed that we also have [`globalTag['data']`](https://github.com/cms-sw/cmssw/blob/594cde36a35054548e86e81ca0fccb1e7fb8ad0a/HLTrigger/Configuration/python/Tools/options.py#L11) currently pointing to a Run-2 GT.
Not sure if that should change already now, or maybe just in the future.

**Comment by Martin-Grunewald** [comment]
Please update now...

**Comment by missirol** [comment]
please test

PR tests should not be affected by the last commit in any way, I think.

**Comment by missirol** [comment]
I tried to inspect the [logs of the `addOnTests`](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-0e19aa/20715/addOnTests/), and I could not find anything problematic.

In Data tests, the L1T and HLT decisions appear consistent with the update of the input file.

Some error messages from `HcalRawToDigi:hcalDigis` have slightly changed; I think these messages were discussed between FOG and HCAL a few weeks ago, and were considered harmless by experts (I have the email exchange, will follow up there).

After the PR, many error/warning messages from modules like `CTPPSPixelRawToDigi:ctppsPixelDigis` and `SiStripRawToDigiModule:siStripDigis` have now vanished.

**Comment by missirol** [comment]
> Some error messages from `HcalRawToDigi:hcalDigis` have slightly changed; I think these messages were discussed between FOG and HCAL a few weeks ago, and were considered harmless by experts (I have the email exchange, will follow up there).

HCAL experts confirmed that the warnings have been fixed by an update in the relevant GTs (see [this and related HN posts](https://hypernews.cern.ch/HyperNews/CMS/get/hcal-performance/1765/1/1/1/1/1/2.html)); they should thus disappear when the HLT `addOnTests` will be updated to run on more recent data.

**Comment by missirol** [comment]
The last commit in this PR cannot have caused the many differences in wf `11634.911`; those are more likely to be related to the issue in https://github.com/cms-sw/cmssw/issues/35109. Tagging @makortel (who had opened that issue).

For this PR, I think we could rely on the results of the [previous comparisons](https://github.com/cms-sw/cmssw/pull/36222#issuecomment-977105338), without having to rerun the tests again, unless others object.

**Comment by makortel** [comment]
> The last commit in this PR cannot have caused the many differences in wf `11634.911`; those are more likely to be related to the issue in #35109. Tagging @makortel (who had opened that issue).

I agree, thanks for notifying.

**Comment by makortel** [comment]
I'd suggest to re-run tests anyway.

**Comment by missirol** [comment]
please test

> I'd suggest to re-run tests anyway.

Okay; thanks for having a look, Matti.

**Comment by missirol** [comment]
+hlt

 - update of the HLT `addOnTests` for collisions Data to use inputs from the 2021 pilot-beam test; this transitions these tests in full to Run 3 in terms of Era, GT, and EDM inputs (before this update, Run-2 input data and GT were used);

 - ad-hoc customisations needed during LS2 for these HLT tests are thus removed where needed;

 - *Note* :  for the moment, the Run-3 Data test of the `HIon` menu will run on 2021 pilot-beam (pp) data, as this is sufficient to ensure that the menu technically works; once Run-3 HI collisions data becomes available, this test will have to be updated with a new EDM input and re-enabling the customisation for the name of the RAW-data input collection (the inputs of other Run-3 Data tests will also likely be updated once actual Run-3 data becomes available).

**Comment by perrotta** [comment]
+1

**Pull request merged**